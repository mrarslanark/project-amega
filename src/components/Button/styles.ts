import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  primary: {
    padding: 12,
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(8, 99, 117, 0.1)',
  },
  secondary: {
    padding: 12,
    borderColor: '#086375',
    backgroundColor: 'rgba(8, 99, 117, 0.1)',
    borderWidth: 2,
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
  },
  tertiary: {
    padding: 12,
    backgroundColor: '#086375',
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
  },
  disabled: {
    padding: 12,
    borderRadius: 12,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0, 0, 0, 0.05)',
  },
});
