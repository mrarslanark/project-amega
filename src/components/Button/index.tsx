import React, {useMemo} from 'react';
import {TouchableOpacity, TouchableOpacityProps} from 'react-native';
import Text from '../Text';
import {styles} from './styles';

interface ButtonProps extends TouchableOpacityProps {
  variant?: 'primary' | 'secondary' | 'tertiary' | 'disabled';
}

const Button: React.FC<ButtonProps> = ({
  variant = 'primary',
  children,
  ...props
}): React.JSX.Element => {
  const selectedVariant = useMemo(() => {
    let currentVariant = variant;
    if (props.disabled) {
      currentVariant = 'disabled';
    }
    return currentVariant;
  }, [props.disabled, variant]);

  return (
    <TouchableOpacity
      activeOpacity={0.8}
      {...props}
      style={styles[selectedVariant]}>
      <Text variant={`button-${selectedVariant}`}>{children}</Text>
    </TouchableOpacity>
  );
};

export default Button;
