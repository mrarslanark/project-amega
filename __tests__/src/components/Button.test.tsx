import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Button from '../../../src/components/Button';

describe('Button', () => {
  it('should render successfully', () => {
    render(<Button>Button</Button>);
    expect(screen).toMatchSnapshot();
  });
});
