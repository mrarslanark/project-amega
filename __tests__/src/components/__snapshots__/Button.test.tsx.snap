// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.tsx should render successfully 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(8, 99, 117, 0.1)",
      "borderRadius": 12,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 12,
    }
  }
>
  <Text
    style={
      {
        "color": "#086375",
        "fontSize": 14,
        "fontWeight": "700",
      }
    }
  >
    Button
  </Text>
</View>
`;
